src/take.ts

```diff
@@ -29,14 +46,18 @@ export function take<T>(array: ReadonlyArray<T>, n: number): Array<T>;
  * @pipeable
  * @category Array
  */
-export function take<T>(n: number): (array: ReadonlyArray<T>) => Array<T>;
+export function take<T extends ReadonlyArray<unknown>, N extends number>(
+  n: N,
+): (array: T) => Take<T, N>;
 
 export function take(...args: ReadonlyArray<unknown>): unknown {
   return purry(takeImplementation, args, lazyImplementation);
 }
 
-const takeImplementation = <T>(array: ReadonlyArray<T>, n: number): Array<T> =>
-  _reduceLazy(array, lazyImplementation(n));
+const takeImplementation = <T extends ReadonlyArray<unknown>, N extends number>(
+  array: T,
+  n: N,
+): unknown => _reduceLazy(array, lazyImplementation(n));
```

Comment:
If we are changing this function with breaking changes lets also change the implementation to use Array.prototype.slice to perform the dataFirst implementation. It's funky that we use our own reducer for this.

Reply:
Is this what you mean? 

Also, what do you think about supporting negative range, same as `slice` while we're at it? I.e. `R.take(arr, -2)` - return last 2 elements of `arr`

Reply:
Would the typing work?

Reply:
Well, been looking at it and I'm not sure we can get them working the same as for taking from he head part. Especially in "headed" cases like `[number, string, ...Array<boolean>]` there really doesn't seem to be a "good" way to define the output array.

However while thinking about this I also realized the same problem exists for regular `take` in "tailed" cases and that my tests do not make sense in those cases either.

Take `const arr = [1, 2] as [...Array<number>, string]`. What should be the output type of `take(arr, n)` for any `n`? Best we could do is maybe something like `Array<number> | [...Array<number>, string]`, or `Array<number | string>`, but neither seems as a good choice?

Reply:
Anyhow if we solve this ^, than I think for `N < 0`, `Take<T, N> = Reverse<Take<Reverse<T>, -N>>` should do the trick?